raiden-cli/src/utils/validation.ts

```diff
@@ -0,0 +1,86 @@
+/* eslint-disable @typescript-eslint/camelcase */
+import { Request, Response, NextFunction } from 'express';
+import {
+  ValidationChain,
+  ValidationError,
+  validationResult,
+  Location,
+  buildCheckFunction,
+} from 'express-validator';
+import _ from 'lodash';
+import { getAddress } from 'ethers/utils';
+
+function formatValdiationErrors(error: ValidationError): string {
+  return `${error.param} - ${error.msg}.`;
+}
+
+export function validate(validations: ValidationChain[]) {
+  return async (request: Request, response: Response, next: NextFunction): Promise<void> => {
+    await Promise.all(validations.map((validation) => validation.run(request)));
+    const errors = validationResult(request).formatWith(formatValdiationErrors);
+
+    if (errors.isEmpty()) {
+      next();
+    } else {
+      response.status(404).json({ errors: errors.array() });
```

Comment:
Since this is a parameter validation error shouldn't that be a 400?

Reply:
I agree with that. The point is that the Raiden API defines it like that. But I think even them are inconsistent in which error code they throw for which invalid parameter. For example do they define [here](https://raiden-network.readthedocs.io/en/latest/rest_api.html#put--api-(version)-tokens-(token_address)) for adding a token, that the API should return a `404`. And they do this all other the place. For URL parameter as well as JSON encoded bodys. So I was basically just following their specification. :man_shrugging:

Reply:
If you agree with the approach to implement the API specification straight forward, then simply close this conversation. Else I keep it open for further discussions.

Reply:
I agree our main target is Raiden API's seamless compatibility, but it'd be worthy putting a FIXME or whatever to discuss proper/better error codes on these.

Reply:
:heavy_check_mark: